### ISSUE TITLE
MySQL

### ISSUE THREAD

**Issue by WesleyVestjens** [magekey/module-bestseller-widget#1]
We are running into errors we traced back to: `MageKey\BestsellerWidget\Block\Widget::getItemsCount()`

**Query:**
```
SELECT
  IF(
    linked.parent_id IS NULL,
    report.product_id,
    linked.parent_id
  ) AS `id`
FROM
  (
    (SELECT
      DATE_FORMAT(period, '%Y') AS `period`,
      SUM(qty_ordered) AS `qty_ordered`,
      `sales_bestsellers_aggregated_yearly`.`product_id`,
      MAX(product_name) AS `product_name`,
      MAX(product_price) AS `product_price`
    FROM
      `sales_bestsellers_aggregated_yearly`
    WHERE (rating_pos <= 5)
      AND (period >= '2018-11-01')
      AND (period <= '2018-11-30')
      AND (store_id IN ('1'))
      AND (store_id IN ('1'))
      AND (1 <> 1)
      AND (period >= '2018-11-01')
      AND (period <= '2018-11-30')
    GROUP BY `period`,
      `product_id`)
    UNION
    ALL
    (SELECT
      DATE_FORMAT(period, '%Y') AS `period`,
      SUM(qty_ordered) AS `qty_ordered`,
      `sales_bestsellers_aggregated_daily`.`product_id`,
      MAX(product_name) AS `product_name`,
      MAX(product_price) AS `product_price`
    FROM
      `sales_bestsellers_aggregated_daily`
    WHERE (period >= '2018-11-01')
      AND (period <= '2018-11-30')
      AND (store_id IN ('1'))
    GROUP BY `product_id`
    ORDER BY `qty_ordered` DESC
    LIMIT 5)
ORDER BY `period` ASC,
  `qty_ordered` DESC
) AS `report`
LEFT JOIN `catalog_product_super_link` AS `linked`
  ON linked.product_id = report.product_id
LIMIT 4
```
**Error Code:** 1055
**Error Message:** Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'deze5242_restore.sales_bestsellers_aggregated_daily.period' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

**Magento version: 2.2.6**

**Comment by magekey** [comment]
@Wesdesignz this issue related to the most of report queries. You can check the query based on refresh bestsellers statistics in admin. You will get the same result.

Here is the trick in Magento 2 core.
https://github.com/magento/magento2/blob/5c82b225c700436ab0f14ad281b025fb0e75b3ad/lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php#L400-L403

If you need to run the query in your sql client just use the `SET SQL_MODE=''`

**Comment by WesleyVestjens** [comment]
Hi @magekey, thanks for replying. The issue arose for us with no products showing up on the main page. My initial response was to debug the query, when I ran into the problem above.

Would manually running the query and checking it be the best action to figure out why no products are shown by the widget?

**Comment by magekey** [comment]
@Wesdesignz I think the best way would be to check the data in tables below:
`sales_bestsellers_aggregated_daily`
`sales_bestsellers_aggregated_monthly`
`sales_bestsellers_aggregated_yearly`

The yealry range was fixed in the last release 2.0.4.
Also use the command below to refresh statistics
`php bin/magento statistics:bestsellers:refresh`

**Comment by WesleyVestjens** [comment]
Alright, I can run the queries in SqlYog now, but I can see something strange:

```
SELECT
  IF(
    linked.parent_id IS NULL,
    report.product_id,
    linked.parent_id
  ) AS `id`
FROM
  (
    (SELECT
      DATE_FORMAT(period, '%Y') AS `period`,
      SUM(qty_ordered) AS `qty_ordered`,
      `sales_bestsellers_aggregated_yearly`.`product_id`,
      MAX(product_name) AS `product_name`,
      MAX(product_price) AS `product_price`
    FROM
      `sales_bestsellers_aggregated_yearly`
    WHERE (rating_pos <= 5)
      AND (period >= '2018-11-01')
      AND (period <= '2018-11-30')
      AND (store_id IN ('1'))
      AND (store_id IN ('1'))
      AND (1 <> 1)
      AND (period >= '2018-11-01')
      AND (period <= '2018-11-30')
    GROUP BY `period`,
      `product_id`)
    UNION
    ALL
    (SELECT
      DATE_FORMAT(period, '%Y') AS `period`,
      SUM(qty_ordered) AS `qty_ordered`,
      `sales_bestsellers_aggregated_daily`.`product_id`,
      MAX(product_name) AS `product_name`,
      MAX(product_price) AS `product_price`
    FROM
      `sales_bestsellers_aggregated_daily`
    WHERE (period >= '2018-11-01')
      AND (period <= '2018-11-30')
      AND (store_id IN ('1'))
    GROUP BY `product_id`
    ORDER BY `qty_ordered` DESC
    LIMIT 5)
ORDER BY `period` ASC,
  `qty_ordered` DESC
) AS `report`
LEFT JOIN `catalog_product_super_link` AS `linked`
  ON linked.product_id = report.product_id
LIMIT 4
```

If it's the yearly aggregation, why is it limited by month? There are orders from September/October, that are not included at the moment, but that is not really a good situation, this way it'll never show statistics if a new period *just* started.

Could you please elaborate on how it's supposed to work?

**Comment by magekey** [comment]
@Wesdesignz please check the comment above.
The yealry range has been fixed in the last release 2.0.4.
